resources/qml/Avatar.qml

```diff
@@ -36,10 +36,40 @@ Rectangle {
         font.pixelSize: avatar.height / 2
         verticalAlignment: Text.AlignVCenter
         horizontalAlignment: Text.AlignHCenter
-        visible: img.status != Image.Ready
+        visible: img.status != Image.Ready && !Settings.useIdenticon
         color: Nheko.colors.text
     }
 
+    Image {
+        id: identicon
+        anchors.fill: parent
+        visible: img.status != Image.Ready && Settings.useIdenticon
+        source: Settings.useIdenticon ? "image://jdenticon/" + userid : ""
+        layer.enabled: true
+
+        MouseArea {
+            anchors.fill: parent
+
+            Ripple {
+                rippleTarget: parent
+                color: Qt.rgba(Nheko.colors.alternateBase.r, Nheko.colors.alternateBase.g, Nheko.colors.alternateBase.b, 0.5)
+            }
+
+        }
+
+        layer.effect: OpacityMask {
+            cached: true
+
+            maskSource: Rectangle {
+                anchors.fill: parent
+                width: avatar.width
+                height: avatar.height
+                radius: Settings.avatarCircles ? height / 2 : 3
+            }
+
+        }
```

Comment:
Can you get rid of those? I noticed they can be quite sluggish on some systems and on some they break avatars completely.

Reply:
As I mentioned the other day, this may be dependent on the jdenticon library. I will look at shaping it in the plugin provider though

Reply:
You are using a QPainter. You can just set a clippath on the painter to make the corners round .-.